cli/tests/unit_node/_fs/_fs_readlink_test.ts

```diff
@@ -0,0 +1,121 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+
+import {
+  assertEquals,
+  assertThrows,
+} from "../../../../test_util/std/testing/asserts.ts";
+import { join } from "node:path";
+import { tmpdir } from "node:os";
+import {
+  mkdirSync,
+  mkdtempSync,
+  readlink,
+  readlinkSync,
+  symlinkSync,
+} from "node:fs";
+import { pathToAbsoluteFileUrl } from "../../unit/test_util.ts";
+
+Deno.test(
+  "[node/fs readLink] read link match target path",
+  () => {
+    const testDir = mkdtempSync(join(tmpdir(), "foo-"));
+    const target = testDir +
+      (Deno.build.os == "windows" ? "\\target" : "/target");
+    const symlink = testDir +
+      (Deno.build.os == "windows" ? "\\symlink" : "/symlink");
+    mkdirSync(target);
+    symlinkSync(target, symlink);
+    readlink(symlink, { encoding: "utf-8" }, async (err, targetPath) => {
+      assertEquals(err, null);
+      assertEquals(targetPath, await target);
+    });
```

Comment:
This call leaks the async op because the internal `Deno.readLink` call is not awaited. We need to surround this call with `new Promise` or use `deferred` object to await the callback triggered.